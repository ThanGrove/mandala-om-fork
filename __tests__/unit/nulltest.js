
const expect = require('expect');
const fs = require('fs');
const path = require('path');
const $ = require('jquery');
const html = fs.readFileSync(path.resolve(__dirname, '../../legacy/index.html'), 'utf8');

jest
    .dontMock('fs')
    .dontMock('jquery');

describe('page', function () {
    beforeEach(() => {
        // console.dir(html);
        document.documentElement.innerHTML = html.toString();
    });

    afterEach(() => {
        // restore the original func after test
        jest.resetModules();
    });

    it('sui-main exists', function () {
        expect($('#sui-main')).toBeTruthy();
        // expect(document.getElementById('#sui-main')).toBeTruthy();
    });

    it('sui search exists', function() {
        expect($('#sui-search')).toBeTruthy();
    });

    it('submit lhasa search', function() {
        $('#sui-search').text('lhasa').click();
        expect(document.documentElement).toMatchSnapshot();
    });

});