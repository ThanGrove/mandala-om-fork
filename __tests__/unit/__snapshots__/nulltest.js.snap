// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page submit lhasa search 1`] = `
<html>
  <head>
    
	
    <meta
      charset="UTF-8"
    />
    
	
    <meta
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
      name="viewport"
    />
    
	
    <script
      src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"
    />
    
	
    <link
      href="https://fonts.googleapis.com/css?family=EB+Garamond:400,400i,500%7COpen+Sans:400,400i,600&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext"
      rel="stylesheet"
    />
    
	
    <script
      src="https://js.arcgis.com/4.12"
    />
    
	
    <script
      src="searchui.js"
    />
    
	
    <script
      src="kmapsSolrUtil.js"
    />
    
	
    <script
      src="pages.js"
    />
    
	
    <script
      src="places.js"
    />
    
	
    <script
      src="subjects.js"
    />
    
	
    <script
      src="audiovideo.js"
    />
    
	
    <script
      src="images.js"
    />
    
	
    <script
      src="texts.js"
    />
    
	
    <script
      src="sources.js"
    />
    
	
    <script
      src="visuals.js"
    />
    
	
    <script
      src="terms.js"
    />
    
	
    <script
      src="collections.js"
    />
    
	
    <link
      href="img/shanti.ico"
      rel="SHORTCUT ICON"
    />
    
	
    <title>
      Mandala
    </title>
    

    <style>
      
	@font-face { font-family: 'shanticon'; src: url('shanticon.ttf?ueun15') format('truetype'); font-weight: normal; font-style: normal; }

    </style>
  </head>
  


  <body>
    
	
    <div
      class="sui-main"
      id="sui-main"
    />
    


    <script>
      

	let suiSite="Mandala";																		// Assume Mandala
	let url=window.location.search;						   										// Get query string
	if (url && url.match(/^\\?b/i))	suiSite="Bhutan";											// Bhutan
	else if (url && url.match(/^\\?c/i))	suiSite="CU";											// Comtemplative University
	var sui=new SearchUI(suiSite,url && url.match(/prod/i));									// Alloc main program, prod or dev
	if (suiSite == "Bhutan") 	sui.ss.project_filter="kmapid: (places-427 subjects-8260)"		// Site specific constant query
	else if (suiSite == "CU")	;
	sui.pages=new Pages();																		// Alloc pages 
	if (url && url.match(/^\\?.e/i))	sui.pages.editing=true;										// 'e' turns on editing
	if (suiSite != "CU") DrawLandingPage();														// Draw Mandala/Bhutan landing

	function DrawLandingPage()																// DRAW SITE SPECIFIC LANDING PAGE
	{
		if (suiSite == "CU") {	
			$("#sui-search").focus();															// Focus on search input
			return;
			}
		let title="Mandala"; 
		if (suiSite == "Bhutan")	title="Bhutan Cultural Library";							// Set title
		$("#sui-results").css({ display:"block", top:"52px" });									// Hide results page	
		$("#sui-headLeft").html(\`&lt;div style='margin-top:8px'&gt;\${title}&lt;/div&gt;\`);					// Set left header
		$("#sui-footer").html("");																// Clear footer
		$("#sui-footer").css("background-color","#ddd");										// Gray
		$("#sui-header").css("background-color","#4d59ca");										// Blue
		$("#sui-results").html("");																// Clear pages
		$("#sui-results").css({"background-size":"100% auto", "background-repeat":"no-repeat", "background-image": "url('https://cicada.shanti.virginia.edu/images/mandala/shanti-image-517836/full/!3000,/0/default.jpg')"});
		let str=\`&lt;div style='text-align:center;width:66%;max-width:800px;margin:12px auto 12px auto'&gt;
		&lt;div style='color:#fff;font-size:24px;margin-bottom:8px;font-weight:700'&gt;
		BHUTAN: A LIVING ARCHIVE&lt;/div&gt;
		&lt;div style='font-size:20px;font-family:"EB Garamond",serif,shanticon; font-weight:400;color:#fff'&gt;
			The Kingdom of Bhutan has vibrant oral and embodied cultures across its mountainous landscape, 
			which are now under pressure from globalization. 
			This project aims to carry out an extensive audio-visual documentation to support local communities.
		&lt;/div&gt;&lt;/div&gt;\`;
		$("#results").append(str.replace(/\\t|\\n|\\r/g,""));									// Remove format and add to div	
		sui.pages.DrawCarousel();
		str=\`&lt;div style='position:absolute;top:calc(100% - 100px); width:66%;max-width:800px;margin:24px auto 12px auto;color:#fff'&gt;	
				&lt;i&gt;The Bhutan Cultural Library is made possible through the contributions and efforts of local individuals 
				and communities in Bhutan in collaboration with Loden Foundation (formerly Shejun Agency) and the University of Virginia. 
				The team gratefully acknowledges the generous support offered by Arcadia throughout the project.&lt;/i&gt;
				&lt;/div&gt;
			&lt;/div&gt;\`;
		$("#results").append(str.replace(/\\t|\\n|\\r/g,""));									// Remove format and add to div	
		sui.ClearQuery();																		// Clear query
//		sui.DrawAdvanced();																		// Refresh advanced search
		$("#sui-search").focus();																// Focus on search input
	}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// HELPERS
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


	$("#sui-header").on("click",(e)=&gt; { if (!e.altKey) return;
		sui.GeoLocate();
	});

	function trace(msg, p1, p2, p3, p4)														// CONSOLE 
	{
		if (p4 != undefined)
			console.log(msg,p1,p2,p3,p4);
		else if (p3 != undefined)
			console.log(msg,p1,p2,p3);
		else if (p2 != undefined)
			console.log(msg,p1,p2);
		else if (p1 != undefined)
			console.log(msg,p1);
		else
			console.log(msg);
	}


    </script>
    



  </body>
</html>
`;
